### DISCUSSION TITLE
Is it `wire:key` or `:wire:key`?

### DISCUSSION THREAD

**Discussion by bastinald** [livewire/livewire#3796] [1. Help] [answered]
According to the docs:

```php
<div>
    @foreach ($users as $user)
        <livewire:user-profile :user="$user" :wire:key="$user->id">
    @endforeach
</div>
```

Is this a typo, or do we need the extra colon in front of `wire`?

**Comment by devcircus** [comment, marked as answer]
It depends on what you are passing as the "key". The same strategy is used by laravel blade components. See the [laravel docs on passing parameters to blade components](https://laravel.com/docs/8.x/blade#passing-data-to-components). "Hard-coded, primative values" may be passed without using the first ":". However, if the value is a variable or uses a php expression, then you must include the first ":".
